skip_night.sc

```diff
@@ -0,0 +1,103 @@
+//////
+// night-skip by Firigion
+// set bed_mode=false; to toggle night skipping via command
+// set bed_mode=true; to have the night be skipped when a player places a bed with 'skip' in its name
+// nights will be skipped as long as the bed exists in the overworld and no other skip bed was placed.
+// This is the default mode.
+//////
+
+bed_mode = true;
+
+global_sleep_time = 12550; // time of day to start skipping
+global_refresh_rate = 100; // how often to check, in ticks
+
+__config() -> {'scope' -> 'global'};
+
+if( bed_mode,
+	
+	( // bed mode
+	__on_player_places_block(player, item_tuple, hand, block) -> (
+		if( player~'dimension' == 'overworld' &&
+			item_tuple:0 ~'_bed' &&  //if it's a bed
+			(name_nbt = item_tuple:2:'display.Name') != null && // and has a custom name
+			lower(parse_nbt(name_nbt):'text') ~ 'skip', // bed has 'skip' in the name
+			
+			_set_bed_block(block, player);
+			
+		);
+	);
+
+	__on_player_breaks_block(player, block) -> (
+		if(global_bed_block && 
+			player~'dimension' == 'overworld' &&
+			(pos(block)==pos(global_bed_block) || 
+			pos(block)==pos_offset(pos(global_bed_block), property(global_bed_block, 'facing'), 1)
```

Comment:
Using `property()` throws deprecation notice in newer carpet versions. Consider the replacement `block_state()` for this line and line 81.